src/byte_stream.rs

```diff
@@ -112,14 +114,39 @@ impl ByteStreamWriter {
             sender,
             metadata_client,
             runtime_handle: handle,
-            event_handle: None,
+            event_handles: vec![],
+            current_offset: 0,
```

Comment:
This is not strictly correct. It is possible that the client is opening on an existing segment.

Reply:
I think we can use an option and do the lazily.

Reply:
Thanks, I see. BTW,  I realize that we can just send a wirecommand to the server and get the offset without maintaining anything in the bytestream writer.

Reply:
I think there is confusion about the original request. The overall requirement for the GStreamer use case is that the application needs to know the offset that a write to will go to. This offset will be used in the index. There are a few ways of achieving this:
Option 1 (preferred, in my opinion): The client tracks the offset. stream_position() returns this client-tracked offset. write() calls are implemented as conditional appends to the expected offset. In the unexpected case that another instance of ByteStreamWriter appends data to the same stream, this conditional append will return an error and the client can choose to panic or seek to the tail.
Option 2: The client tracks the offset. stream_position() returns this client-tracked offset. write() calls are normal appends.   In the unexpected case that another instance of ByteStreamWriter appends data to the same stream, the client-tracked offset will be incorrect forever.
Option 3: The client does not track the offset. stream_position() queries the server for the current tail and is expensive. write() calls are normal appends. In the unexpected case that another instance of ByteStreamWriter appends data to the same stream, then stream_position() may or may not return the offset of the next append (this is a race condition).

Reply:
Sounds like in this case we need to use `ConditionalAppend` wirecommand and have something similar to the `ConditionalOutputStream` .

Reply:
@Tristan1900 Yes. I think we can probably have a either a flag or a "wireCommandFactory" passed into `SegmentReactor` which the EventStream sets to `AppendBlockEnd` and the ByteStream sets to `ConditionalAppend`